src/util/change-types.ts

```diff
@@ -69,11 +68,30 @@ export const changeStationsTypeInBatch = (
             changeStationType(graph, stnId, newStnType);
         });
 
+/**
+ * Change selected stations' type to newStnType in batch.
+ * @param graph Graph.
+ * @param selected Selected
+ * @param newStnType New station's type.
+ * @returns Nothing.
+ */
+export const changeStationsTypeSelected = (
```

Comment:
`changeStationsTypeInBatch` is a special case of this method. Try merge into one. You may select all here using a boolean switch or passing all the elements from modal.